lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -276,24 +408,50 @@ def with_spectrum(self, power, num_points_per_block=1, variance=None) -> "PowerS
         if len(power) != len(self.power):
             raise ValueError("New power spectral density vector has incorrect length")
 
-        ps = copy(self)
-        ps.power = power
-        ps.num_points_per_block = num_points_per_block
-        ps._variance = variance
-
-        return ps
+        return PowerSpectrum(
+            self.frequency,
+            power,
+            self.sample_rate,
+            self.total_duration,
+            self.unit,
+            num_points_per_block=num_points_per_block,
+            total_samples_used=self.total_samples_used,
+            variance=variance,
+        )
 
-    def plot(self, **kwargs):
+    def plot(self, *, show_excluded=False, **kwargs):
         """Plot power spectrum
 
         Parameters
         ----------
+        show_excluded : bool
+            Show ranges that were excluded from fitting.
         **kwargs
             Forwarded to :func:`matplotlib.pyplot.plot`.
         """
         import matplotlib.pyplot as plt
 
-        plt.plot(self.frequency, self.power, **kwargs)
+        if show_excluded:
+            if self._downsampling_factor > 1:
+                plt.plot(self.unfiltered_frequency, self.unfiltered_power, label="_")
```

Comment:
I had no idea that you could exclude labels from the legend like this 🤯

Reply:
Yeah. It's pretty nice. :)